pkg/validation/providers.go

```diff
@@ -0,0 +1,83 @@
+package validation
+
+import (
+	"fmt"
+	"io/ioutil"
+
+	"github.com/oauth2-proxy/oauth2-proxy/v7/pkg/apis/options"
+)
+
+// validateProviders is the initial validation migration for multiple providrers
+// It currently includes only logic that can verify the providers one by one and does not break the valdation pipe
+func validateProviders(o *options.Options) []string {
+	msgs := []string{}
+
+	// validate general multiple provider configuration
+	if len(o.Providers) == 0 {
+		msgs = append(msgs, "at least one provider has to be defined")
+	}
+	if o.SkipProviderButton && len(o.Providers) > 1 {
+		msgs = append(msgs, "SkipProviderButton and multiple providers are mutually exclusive")
+	}
+
+	providerIDs := make(map[string]string)
```

Comment:
Bit of a nit, but can we copy the pattern in upstreams and use `map[string]struct{}`? Just means changing the declaration here and when added, will add suggestions
```suggestion
	providerIDs := make(map[string]struct{})
```

Reply:
Sure, I'll just verify it does not mess up any tests..